layers/ip6.go

```diff
@@ -280,165 +229,191 @@ func decodeIPv6(data []byte, p gopacket.PacketBuilder) error {
 	return p.NextDecoder(ip6.NextHeader)
 }
 
-func (i *IPv6HopByHop) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
-	var bytes []byte
-	var err error
+type IPv6HeaderTLVOption interface {
```

Comment:
> It appears to me that the interface must escape such a function because of IPv6AlignHeaderTLVOptions(options *[]IPv6HeaderTLVOption)? Or ipv6AlignHeaderTLVOptions(), same thing...

Bad example, I think, but yeah, decodeIPv6HeaderTLVOption()...